components/profile-controller/pkg/controller/profile/profile_controller.go

```diff
@@ -52,6 +52,22 @@ const USER = "user"
 const ROLE = "role"
 const ADMIN = "admin"
 
+// Kubeflow default role names
+// TODO: Make kubeflow roles configurable (krishnadurai)
+// This will enable customization of roles.
+const(
+	kubeflowAdmin =	"kubeflow-admin"
+	kubeflowEdit =	"kubeflow-edit"
+	kubeflowView =	"kubeflow-view"
+)
+
+//roleBindingNameMap maps frontend role names to k8s role names
```

Comment:
Again, this seems very weird to me.
The frontend is a client of this API, so it's as if we are letting the client decide what the API is, instead of defining it here.

This brings me to a second observation.
There are 2 places where binding creation takes place.
1. The profile-controller
2. The KFAM API

I believe we should standardize and do it in one place only.
In this case, should the profile-controller also use the KFAM API to create a binding?
I think this will also reduce the need to define the same names in different places.
/cc @kunmingg @jlewi

Reply:
@yanniszark I agree with your view and I think we should address this. The way I'm thinking this can be done is through making Kfam the sole owner of creating, listing and deleting the RoleBinding resources. Likewise when we should be able to make these roles more composable, I think we could store these role definitions or mappings in a storage instead - this is why I've marked a TODO.
Given the scope of what we are trying to achieve in this PR I think these changes should not halt this PR.

Reply:
I'll clarify here that the client is not defining the back-end API parameter values. I think the values used by the back-end API may or should not be the same as what the cluster role names are because the names which we use might clash with other applications in K8s as well. Here I feel the values 'admin', 'edit' and 'view' should be the values accepted by the API whereas the bindings may have a 'kubeflow' prefix.